src/view/components/TabManager/index.tsx

```diff
@@ -0,0 +1,136 @@
+import * as React from 'react';
+import { connect, MapStateToProps, MapDispatchToProps, DispatchProp } from 'react-redux';
+import { Text, Tab as NativeTab, Tabs as NativeTabs } from 'native-base';
+
+import { Tab, TabList, ElementTab, ElementTabList, StoreTabList } from '../../../store/modules/tabs/types';
+import tabActions from '../../../store/modules/tabs/actions';
+import { State } from '../../../store/types';
+
+/** TabManager OwnProps */
+interface OwnProps {}
+
+interface StoreProps {
+  activeTab: string;
+}
+
+interface DispatchProps extends DispatchProp<State> {
+  setTabs(tabs: ElementTabList, activeTab: ElementTab): void;
+  setActiveTab(activeTab: ElementTab): void;
+  clearTabs(): void;
+}
+
+interface OwnState {}
+
+/** TabManager props */
+export type Props = OwnProps & StoreProps & DispatchProps;
+
+function createSimpleComponent(tabs: ElementTabList) {
+
+  /**
+  * Page that renders different component depending on which tab the user is on
+  *
+  * Should not be instantiated directly, only extended. Unable to mark abstract due to react-redux
+  */
+  return class TabManager extends React.Component<Props, OwnState> {
+
+    /************************* Member Variables ************************/
+    private tabs: ElementTabList;
+
+    /************************* Member Functions ************************/
+
+    constructor(props: Props) {
+      super(props);
+
+      // Validation
+      if (!tabs.length) {
+        throw new Error('Cannot create TabNavigator with no tabs');
+      }
+
+      // Initialization
+      this.tabs = tabs;
+    }
+
+    private createTab(tab: ElementTab) {
+      return <NativeTab heading={tab.name} key={tab.name}>{tab.element()}</NativeTab>;
```

Comment:
are the tabs scrollable?

Reply:
No I have opened up an issue against NativeBase regarding this because it seems the scrollable tabs are not in the typescript declaration file. I wanted to make it so if there were more than 4 tabs it would scroll

Reply:
ahh I see. Looks like they've started working on it! :D https://github.com/GeekyAnts/NativeBase/issues/1517